File: src/test/java/com/widehouse/whatnext/domain/TaskTest.java
package com.widehouse.whatnext.domain;

import static com.widehouse.whatnext.domain.TaskStatus.DONE;
import static com.widehouse.whatnext.domain.TaskStatus.TODO;
import static org.assertj.core.api.BDDAssertions.then;

import org.junit.Before;
import org.junit.Test;

public class TaskTest {
    private Category category;

    @Before
    public void setup() {
        category = new Category(1, "home", "ffffff");
    }

    @Test
    public void testConstructor() {
        Task task = new Task("desc", 1, TODO, category);

        then(task)
                .hasFieldOrPropertyWithValue("description", "desc")
                .hasFieldOrPropertyWithValue("priority", 1)
                .hasFieldOrPropertyWithValue("status", TODO)
                .hasFieldOrPropertyWithValue("category", category);
    }

    @Test
    public void testEquals() {
        Task x = new Task("desc", 1, TODO, category);
        Task y = new Task("desc", 1, TODO, category);

        then(x.equals(y) && y.equals(x))
                .isTrue();
    }

    @Test
    public void setStatus_thenUpdateStatus() {
        Task task = new Task("desc", 1, TODO, category);

        task.setStatus(DONE);

        then(task.getStatus())
                .isEqualTo(DONE);
    }
}

File: src/test/kotlin/com/widehose/whatnext/domain/TaskTest.kt
package com.widehose.whatnext.domain

import com.widehouse.whatnext.domain.Category
import com.widehouse.whatnext.domain.Task
import com.widehouse.whatnext.domain.TaskStatus
import org.assertj.core.api.BDDAssertions
import org.junit.Test
import java.time.ZonedDateTime

class TaskTest {
    @Test
    fun testCodeCoverageForTaskClass() {
        val tested = Task(1, "desc", 1, TaskStatus.TODO, Category(1, "category", ""), ZonedDateTime.now()).apply {
            BDDAssertions.then(this.toString())
                    .contains("id=1")
                    .contains("description=desc")
                    .contains("status=TODO")
                    .contains("category=")
                    .contains("createdDate=")

            id = 2
            BDDAssertions.then(id).isEqualTo(2)
            description = "desc2"
            BDDAssertions.then(description).isEqualTo("desc2")
            priority = 2
            BDDAssertions.then(priority).isEqualTo(2)
            val testHour = ZonedDateTime.now().plusHours(10)
            createdDate = testHour
            BDDAssertions.then(createdDate).isEqualTo(testHour)
        }

        val testedCopy = tested.copy(id = 3).apply {
            BDDAssertions.then(this.equals(tested))
                    .isFalse()
            BDDAssertions.then(this.hashCode())
                    .isNotEqualTo(tested.hashCode())
        }
    }
}